webui/react/src/components/FilterForm/components/type.ts

```diff
@@ -0,0 +1,108 @@
+import { V1ColumnType, V1LocationType, V1ProjectColumn } from 'services/api-ts-sdk';
+import { ValueOf } from 'shared/types';
+
+export const FormKind = {
+  Field: 'field',
+  Group: 'group',
+} as const;
+
+export type FormKind = ValueOf<typeof FormKind>;
+
+export type FormFieldValue = string | number | null;
+
+export type FormField = {
+  readonly id: string;
+  readonly kind: typeof FormKind.Field;
+  columnName: V1ProjectColumn['column'];
+  location: V1LocationType;
+  type: V1ColumnType;
```

Comment:
do we need `location` and `type` in the field?

Reply:
Filter API requires these